src/Select.js

```diff
@@ -415,13 +425,20 @@ const Select = React.createClass({
 
 	handleInputFocus (event) {
 		if (this.props.disabled) return;
-		var isOpen = this.state.isOpen || this._openAfterFocus || this.props.openOnFocus;
+		const isOpen = this.state.isOpen || this._openAfterFocus || this.props.openOnFocus;
 		if (this.props.onFocus) {
 			this.props.onFocus(event);
 		}
+
+		// To make the react-select act more natively, render the input value with the corresponding label on focus. However, we 
+		// cannot do this for custom valueRenderer's bc the label can be manipulated during render (e.g. option.label.toUpperCase())
+		// so the option labels could be inconsistent.
+		let inputLabel = this.getInputLabel();
```

Comment:
Also it seems like users with valueRenderer should be able to benefit from this. Maybe we can warn users against manipulating the value?